### PR TITLE
Use stored passphrase in device provisioning flow + store passphrase on signup

### PR DESCRIPTION
**Pull request by chromakode**
I started this branch looking into @cjb's report that he was asked for his passphrase when trying to add a new device. There were several things going on:

* `DeviceAdd` requests the passphrase explicitly before starting the kexp flow to avoid two colliding UI prompts. However, it wasn't using the `GetPassphraseStream` variant that checked the secret store.
* We weren't storing passphrases in the secret store on signup, so test users wouldn't have the secret in the secret store until they logged in again.
* `GetPassphraseStreamExport` would bail if the secret store was missing the item instead of falling back to a passphrase entry.

I've attempted to fix these issues in a way that makes sense to me, but I'm not sure if they fit the organization style of the rest of the Go codebase. Since it seems like `	GetPassphraseStreamExport` is only used in this spot, I refactored it into what I think is a more self-explanatory name: `GetPassphraseStreamStored`. I also dropped the exporting behavior to make it convenient to pass the `PassphraseStream` to `NewKex2Provisioner`. However, there are many ways to skin this cat; feedback appreciated!

@maxtaco @patrickxb @keybase/react-hackers

### CODE REVIEW

File: go/libkb/login_state.go [hunk 1]
```diff
@@ -296,28 +297,29 @@ func (s *LoginState) GetPassphraseStreamExport(ui SecretUI) (pps keybase1.Passph
...
+			}
+			stream, err := NewPassphraseStreamLKSecOnly(secret)
+			if err != nil {
+				return pps, err
+			}
+			stream.SetGeneration(lks.Generation())
+			s.G().Log.Debug("| got passphrase stream from secret store")
+			return stream, nil
+		} else {
+			s.G().Log.Debug("| failed to get passphrase stream from secret store: %s", err)
```
Comment (patrickxb) [inline go/libkb/login_state.go]: Can we add a test that hits this path (that used to fail before)?

**Comment by keybase-ci-visdiff** [comment]
The commits 95b0a8aa6f38...8d377247385d introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475780073934-95b0a8aa6f38-8d377247385d-linux/New_Proof_PGP_generating-Generating_PGP-95b0a8aa6f38.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475780073934-95b0a8aa6f38-8d377247385d-linux/New_Proof_PGP_generating-Generating_PGP-8d377247385d.png)  
   New_Proof_PGP_generating-Generating_PGP comparison
</summary>

**Comment by maxtaco** [comment]
Seems like this PR is causing #4661 and #4634.  Investigating further...

**Comment by maxtaco** [comment]
Confirmed via bisection

**Pull request merged**